The sizeof() operator includes the NUL terminator, so the comments are wrong.

--- Linux-PAM-0.77/modules/pam_filter/pam_filter.c	2003-06-27 19:06:37.000000000 -0400
+++ Linux-PAM-0.77/modules/pam_filter/pam_filter.c	2003-06-27 19:07:20.000000000 -0400
@@ -150,7 +150,7 @@
 
 	/* the "ARGS" variable */
 
-#define ARGS_OFFSET    5                          /*  sizeof('ARGS=');  */
+#define ARGS_OFFSET    5                          /*  strlen('ARGS=');  */
 #define ARGS_NAME      "ARGS="
 
 	size += ARGS_OFFSET;
@@ -174,7 +174,7 @@
 
 	/* the "SERVICE" variable */
 
-#define SERVICE_OFFSET    8                    /*  sizeof('SERVICE=');  */
+#define SERVICE_OFFSET    8                    /*  strlen('SERVICE=');  */
 #define SERVICE_NAME      "SERVICE="
 
 	pam_get_item(pamh, PAM_SERVICE, (const void **)&tmp);
@@ -196,7 +196,7 @@
 
 	/* the "USER" variable */
 
-#define USER_OFFSET    5                          /*  sizeof('USER=');  */
+#define USER_OFFSET    5                          /*  strlen('USER=');  */
 #define USER_NAME      "USER="
 
 	pam_get_user(pamh, &tmp, NULL);
@@ -222,7 +222,7 @@
 
 	/* the "USER" variable */
 
-#define TYPE_OFFSET    5                          /*  sizeof('TYPE=');  */
+#define TYPE_OFFSET    5                          /*  strlen('TYPE=');  */
 #define TYPE_NAME      "TYPE="
 
 	size = TYPE_OFFSET+strlen(type);
